course/include/Isometry.hpp

```diff
@@ -151,5 +159,74 @@ inline std::ostream& operator<<(std::ostream& os, const Matrix3& mat) {
               << "]]";
 }
 
+const inline Matrix3 matrix_multiplication(Matrix3 mat, Matrix3 rhs) {
+    Matrix3 result;
+    for (uint16_t row = 0; row < 3; ++row) {
+        for (uint16_t column = 0; column < 3; ++column) {
+            result[row][column] =
+                mat[row][0] * rhs[0][column] + mat[row][1] * rhs[1][column] + mat[row][2] * rhs[2][column];
+        }
+    }
+    return result;
+}
+
+class Isometry {
+   public:
+    Isometry(const Matrix3& rotation);
+    Isometry(const Vector3& translation, const Matrix3& rotation);
+
+    Isometry compose(const Isometry& isometry) const;
+    Isometry inverse() const;
+    Matrix3 rotation() const;
+    Vector3 transform(Vector3 translation) const;
+    Vector3 translation() const;
+
+    static Isometry FromTranslation(const std::initializer_list<double>& values);
+    static Isometry FromEulerAngles(double yaw, double pitch, double roll);
+    static Matrix3 RotateAround(const Vector3& axis, double angle);
+
+    // Operators
+    bool operator==(const Isometry& a) const;
+    Vector3 operator*(const Vector3& a) const;
+    Isometry operator*(const Isometry& a) const;
+
+   private:
+    Matrix3 rotation_;
+    Vector3 translation_;
+};
+
+inline ::testing::AssertionResult areAlmostEqual(const Isometry& lhs, const Isometry& rhs, const double tolerance) {
+    auto lhs_rotation = lhs.rotation();
+    auto lhs_translation = lhs.translation();
+
+    auto rhs_rotation = rhs.rotation();
+    auto rhs_translation = rhs.translation();
+
+    for (uint8_t index_x = 0; index_x < 2; ++index_x) {
```

Comment:
This loop will never iterate over the elements in row 2. Same below for columns.